packages/vite/src/node/plugins/html.ts

```diff
@@ -294,12 +295,16 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
         seen: Set<string> = new Set()
       ): HtmlTagDescriptor[] => {
         const tags: HtmlTagDescriptor[] = []
-        chunk.imports.forEach((file) => {
-          const importee = bundle[file]
-          if (importee && importee.type === 'chunk') {
-            tags.push(...getCssTagsForChunk(importee, seen))
-          }
-        })
+        if (!analyzedChunk.has(chunk)) {
+          analyzedChunk.set(chunk, 1)
+          chunk.imports.forEach((file) => {
+            const importee = bundle[file]
+            if (importee && importee.type === 'chunk') {
```

Comment:
**suggestion (non-blocking):** This may improved with optional chaining `if (importee?.type === 'chunk') {`

_unless `importee` is `boolean | OtherType`, I don't see that in GitHub Web UI_

Reply:
So I checked in VSCode and it's of type `OutputAsset | OutputChunk`
So optional chaining is possible here 🙂 

```suggestion
            if (importee?.type === 'chunk') {
```